crates/reliable-broadcast/src/lib.rs

```diff
@@ -92,16 +107,37 @@ where
                             .await,
                     )
                 }
+                .boxed()
             };
             let message: Req = message.into();
+            let (mut fut_tx, fut_rx) = futures_channel::mpsc::channel(receivers.len());
             for receiver in receivers {
-                fut.push(send_message(receiver, message.clone(), None));
+                fut_tx
+                    .try_send(send_message(receiver, message.clone(), None))
+                    .expect("must enqueue");
             }
-            while let Some((receiver, result)) = fut.next().await {
-                match result.and_then(|msg| msg.try_into().map_err(|e| anyhow::anyhow!("{:?}", e)))
-                {
-                    Ok(ack) => {
-                        if let Ok(Some(aggregated)) = aggregating.add(receiver, ack) {
+            let mut concurrent_verify_stream = concurrent_map(
```

Comment:
not sure if it's just me, this looks a bit ugly. I was thinking about something like this
```
  network_futures = vec![];
  aggregation_futures = vec![];
  select! {
    result = network_futures.next() {
      aggregation_futures.push(bounded_executor.spawn(|| aggregating.add(..)).await);
    }
    result = aggregation_futures.next() {
      if aggregated {
        return;
      }
    }
  }
```

Reply:
it does. it's my obsession with `concurrent_map` at this point. Cleaned it up and used loop { select! } now.